p2panda-auth/Cargo.toml

```diff
@@ -16,6 +16,9 @@ keywords = ["auth", "access-control", "groups"]
 all-features = true
 
 [dependencies]
+petgraph = { version = "0.8.1", features = ["serde-1"] }
+rand = { version = "0.9.1", features = ["alloc"] }
```

Comment:
`rand` is used in the test utils. I suggest you introduce a `test_utils` feature flag and restrict these features to that flag and `test` environments, then you can also make this dependency optional

Reply:
Alternatively you can move `rand` just into dev-dependencies, but I assume you want to fuzz your code at one point, then it makes sense to already expose the `test_utils`